src/Tools/dotnet-user-jwts/src/Commands/CreateCommand.cs

```diff
@@ -0,0 +1,204 @@
+                CommandOptionType.SingleValue);
+
+            var expiresOnOption = cmd.Option(
+                "--expires-on",
+                @"The UTC date & time the JWT should expire in the format 'yyyy-MM-dd [[[[HH:mm]]:ss]]'. Defaults to 6 months after the --not-before date. " +
+                         "Do not use this option in conjunction with the --valid-for option.",
+                CommandOptionType.SingleValue);
+
+            var validForOption = cmd.Option(
+                "--valid-for",
+                "The period the JWT should expire after. Specify using a number followed by a period type like 'd' for days, 'h' for hours, " +
+                         "'m' for minutes, and 's' for seconds, e.g. '365d'. Do not use this option in conjunction with the --expires-on option.",
+                CommandOptionType.SingleValue);
+
+            cmd.HelpOption("-h|--help");
+
+            cmd.OnExecute(() =>
+            {
+                var (options, isValid) = ValidateArguments(
+                    app.ProjectOption, schemeNameOption, nameOption, audienceOption, issuerOption, notBeforeOption, expiresOnOption, validForOption, rolesOption, scopesOption, claimsOption);
+
+                if (!isValid)
+                {
+                    return 1;
+                }
+
+                return Execute(app.ProjectOption.Value(), options);
+            });
+        });
+    }
+
+    private static (JwtCreatorOptions, bool) ValidateArguments(
+        CommandOption projectOption,
+        CommandOption schemeNameOption,
+        CommandOption nameOption,
+        CommandOption audienceOption,
+        CommandOption issuerOption,
+        CommandOption notBeforeOption,
+        CommandOption expiresOnOption,
+        CommandOption validForOption,
+        CommandOption rolesOption,
+        CommandOption scopesOption,
+        CommandOption claimsOption)
+    {
+        var isValid = true;
+        var project = DevJwtCliHelpers.GetProject(projectOption.Value());
+        var scheme = schemeNameOption.HasValue() ? schemeNameOption.Value() : "Bearer";
+        var name = nameOption.HasValue() ? nameOption.Value() : Environment.UserName;
+
+        var audience = audienceOption.HasValue() ? audienceOption.Value() : DevJwtCliHelpers.GetApplicationUrl(project);
+        if (audience is null)
+        {
+            Console.WriteLine("Could not determine the project's HTTPS URL. Please specify an audience for the JWT using the --audience option.");
+            isValid = false;
+        }
+        var issuer = issuerOption.HasValue() ? issuerOption.Value() : DevJwtsDefaults.Issuer;
+
+        var notBefore = DateTime.UtcNow;
+        if (notBeforeOption.HasValue())
+        {
+            if (!ParseDate(notBeforeOption.Value(), out notBefore))
+            {
+                Console.WriteLine(@"The date provided for --not-before could not be parsed. Ensure you use the format 'yyyy-MM-dd [[[[HH:mm]]:ss]]'.");
```

Comment:
I think the example format here is needlessly complex (my fault I'm sure). The square braces are supposed to represent optionality, so in this case it could just be `yyyy-MM-dd [HH:mm[:ss]]` right? i.e.
- the date portion is required
- the whole timestamp is optional
- if the timestamp is supplied, the seconds portion is optional

Reply:
I removed having seconds as a parseable value. I don't think folks we'll get that granular with their validate timespans. If they do care, it's better to give than to take away.